File: 151002/Protchenko/lab1/rest-api/src/main/java/by/bsuir/restapi/repository/PostRepository.java
package by.bsuir.restapi.repository;

import by.bsuir.restapi.model.entity.Post;
import by.bsuir.restapi.repository.base.InMemoryRepository;
import org.springframework.stereotype.Repository;

@Repository
public class PostRepository extends InMemoryRepository<Post> {
}

File: 151004/Ivanenko/DC_Lab3/DC_Lab3/Publisher/Repository/Implementation/PostRepository.cs
using Publisher.Entity.Db;
using Publisher.Repository.Implementation.Common;
using Publisher.Repository.Interface;
using Publisher.Storage.Common;

namespace Publisher.Repository.Implementation
{
    public class PostRepository(DbStorage context) : AbstractCrudRepository<Post>(context), IPostRepository
    {
        private readonly DbStorage _context = context;

        public override Post Add(Post post)
        {
            var tweet = new Tweet { Id = post.TweetId };
            _context.Tweets.Attach(tweet);
            post.Tweet = tweet;

            return base.Add(post);
        }

        public override async Task<Post> AddAsync(Post post)
        {
            var tweet = new Tweet { Id = post.TweetId };
            _context.Tweets.Attach(tweet);
            post.Tweet = tweet;

            return await base.AddAsync(post);
        }
    }
}
